src/main/java/com/academy/edge/studentmanager/services/impl/GradeServiceImpl.java

```diff
@@ -48,4 +50,28 @@ public GradeResponseDTO saveGrade(GradeCreateDTO gradeCreateDTO) {
 
         return modelMapper.map(grade, GradeResponseDTO.class);
     }
+
+    @Override
+    @Transactional
+    public HttpStatus deleteGrade(GradeDeleteDTO gradeDeleteDTO) {
+        Integer deleted;
+        try{
+            Student student = studentRepository
+                    .findByEmail(gradeDeleteDTO.getStudentEmail())
+                    .orElseThrow(() -> new ResponseStatusException(NOT_FOUND, "Student not found"));
+
+            deleted = gradeRepository.deleteGradeByStudentIdAndSubjectCodeAndPeriod(
+                                student.getId(),
+                                gradeDeleteDTO.getSubjectCode(),
+                                gradeDeleteDTO.getPeriod());
+        }
+        catch(Exception e){
+            throw new RuntimeException(e);
+        }
+
+        if(deleted == 1){
+            return HttpStatus.NO_CONTENT;
+        }
+        return HttpStatus.NOT_FOUND;
+    }
```

Comment:
Eu acho que o try aqui não tem muita eficácia e deixa o código menos legível.
Quando não há o estudante, ele já lança a exceção corretamente que é reconhecida pelo exception handler do spring.

Imagino que possamos retornar sempre o resultado do repositório.
Outro fator importante também é sobre o método ser transacional, creio que não há muito efeito.

```suggestion
    @Override
    public boolean deleteGrade(GradeDeleteDTO gradeDeleteDTO) {

            Student student = studentRepository
                    .findByEmail(gradeDeleteDTO.getStudentEmail())
                    .orElseThrow(() -> new ResponseStatusException(NOT_FOUND, "Student not found"));

            return gradeRepository.deleteGradeByStudentIdAndSubjectCodeAndPeriod(
                                student.getId(),
                                gradeDeleteDTO.getSubjectCode(),
                                gradeDeleteDTO.getPeriod());
   }
```